### PR TITLE
ARROW-12444: [Rust] Remove rust

### PR DESCRIPTION
**Pull request by jorgecarleitao**
Removes rust from this repository and uses git clone to fetch the arrow-rs repo for integration tests.

Each commit is an independent change.

Note: Rust implementations have moved to:
* https://github.com/apache/arrow-rs
* https://github.com/apache/arrow-datafusion

### CODE REVIEW

File: dev/archery/archery/lang/rust.py
Comment (alamb) [inline dev/archery/archery/lang/rust.py:1]: I am not sure this one should be removed @jorgecarleitao  -- it might be how rust is run in the integration test framework.

**Review by alamb — APPROVED** [review]
The basic idea looks good to me @jorgecarleitao. Thank you. 

The only thing I think may be worth a second set of eyes is the archery integration stuff (but perhaps that is because I don't have a great handle on the integration test plan)

**Review by nevi-me — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
Ok, I think it is time to merge this PR @jorgecarleitao

**Comment by jorgecarleitao** [comment]
I think that we agreed over the mailing list to not merge this until @kszucs gives green light, post release.

I suggest that we take the time to complete the release, confirm that we are in a new development cycle, and then merge this. The rational being that I would like to reduce the risk of pushing merge conflicts to the release manager.

**Comment by kszucs** [comment]
Yes, please hold off until we merge #10165.

**Review by andygrove — APPROVED** [review]

**Comment by jorgecarleitao** [comment]
Rebased :)

File: dev/archery/archery/cli.py
Comment (jorgecarleitao) [inline dev/archery/archery/cli.py:644]: This should stay

**Comment by kszucs** [comment]
@jorgecarleitao @andygrove @alamb I think we can start to roll-up the rust pull requests now. There are a couple of open ones remaining, I assume we should close those and merge this one.

**Comment by kszucs** [comment]
Rust releated part should be removed from `ci/docker/linux-apt-lint.dockerfile`.

There is a reference for the rust implementation in `matlab/doc/matlab_interface_for_apache_arrow_design.md` which 
should be updated.

I assume the rust post release script should be removed as well `ci/release/post-07-rust.sh` since the new releases will be cut from `arrow-rs`.

There is a `debian-rust` configuration in the docker compose as well as in the crossbow tasks, but I guess we can keep that around to exercise the rust build script for the integration tests on a nightly basis.

**Comment by jorgecarleitao** [comment]
Thanks @kszucs . I have applied all those changes.

**Comment by alamb** [comment]
Thanks @jorgecarleitao

**Review by kszucs — APPROVED** [review]
Thanks everyone! I'm merging it, hopefully we didn't forget anything!

**Comment by alamb** [comment]
Thank you @kszucs

**Pull request closed without merging**
